aws/resource_aws_pinpoint_event_stream.go

```diff
@@ -0,0 +1,110 @@
+package aws
+
+import (
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/pinpoint"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsPinpointEventStream() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsPinpointEventStreamUpsert,
+		Read:   resourceAwsPinpointEventStreamRead,
+		Update: resourceAwsPinpointEventStreamUpsert,
+		Delete: resourceAwsPinpointEventStreamDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"application_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"destination_stream_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"role_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceAwsPinpointEventStreamUpsert(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).pinpointconn
+
+	applicationId := d.Get("application_id").(string)
+
+	d.SetId(applicationId)
+
+	params := &pinpoint.WriteEventStream{}
+
+	if d.HasChange("destination_stream_arn") {
+		params.DestinationStreamArn = aws.String(d.Get("destination_stream_arn").(string))
+	}
+
+	if d.HasChange("role_arn") {
+		params.RoleArn = aws.String(d.Get("role_arn").(string))
+	}
+
+	req := pinpoint.PutEventStreamInput{
+		ApplicationId:    aws.String(applicationId),
+		WriteEventStream: params,
+	}
+
+	_, err := conn.PutEventStream(&req)
+	if err != nil {
+		return err
```

Comment:
It might be helpful for operators to have some context about what failed, e.g.

```go
return fmt.Errorf("error putting event stream (%s): %s", applicationId, err)
```